<launch>

  <arg name="LAUNCH_DYNAMIC_TF_PUBLISHER" default="true" />
  <arg name="default_frame_id" default="odom_on_ground" />
  <arg name="marker_tf_prefix" default="/jsk_model_marker_interface/robot"/>

  <arg name="PUBLISH_CAMERA_INFO" default="false" />
  <arg name="PUBLISH_REACHABLE_MAP" default="false" />

  <!-- robot specific setting -->
  <arg name="robot_model_config" default="$(find drc_task_common)/config/robot_interactive_marker.yaml" />

  <!-- robot marker -->
  <include file="$(find jsk_interactive_marker)/launch/out_of_body_marker.launch">
    <arg name="LAUNCH_DYNAMIC_TF_PUBLISHER" value="$(arg LAUNCH_DYNAMIC_TF_PUBLISHER)" />
    <arg name="default_frame_id" value="$(arg default_frame_id)" />
    <arg name="robot_marker_frame_id" value="robot_marker_root" />
    <arg name="use_default_robot_model_config" value="false" />
    <arg name="robot_model_config" value="$(arg robot_model_config)" />
    <arg name="ROBOT_DESCRIPTION" value="/ocs/robot_description" />
  </include>

  <group if="$(arg PUBLISH_CAMERA_INFO)">
    <!-- publish carmine tf -->
    <node pkg="tf" type="static_transform_publisher" name="static_transform_publisher_for_carmine" 
          args="0.091965 0.005659 0.111169 -0.563491677276 0.575071457361 -0.420579296396 0.418190152935 $(arg marker_tf_prefix)/HEAD_LINK1 $(arg marker_tf_prefix)/CARMINE 50" />
    <node name="static_transform_publisher_for_camera_link" pkg="tf" type="static_transform_publisher"
          args="-0.045 0.000 0.000 0.500 -0.500 0.500 0.500 $(arg marker_tf_prefix)/CARMINE $(arg marker_tf_prefix)/camera_link 50"/>

    <group ns="interactive_marker_carmine">
      <include file="$(find rgbd_launch)/launch/kinect_frames.launch">
        <arg name="camera" value="camera"/>
        <arg name="tf_prefix" value="$(arg marker_tf_prefix)"/>
      </include>
    </group>

    <!-- publish camera info  -->
    <node pkg="jsk_rviz_plugins" type="relay_camera_info.py" name="relay_camera_info">
      <param name="frame_id" value="$(arg marker_tf_prefix)/camera_depth_optical_frame"/>
      <remap from="input" to="/camera/depth_registered/camera_info"/>
      <remap from="output" to="$(arg marker_tf_prefix)/camera/depth_registered/camera_info"/>
    </node>
  </group>

  <!-- publish reachable map -->
  <group if="$(arg PUBLISH_REACHABLE_MAP)">
    <node pkg="drc_task_common" type="reachability-map-publisher.l" name="reachability_map_publisher" />
  </group>

</launch>
